### PR TITLE
Creating a vscode configuration for debugging the server project

### LINKED ISSUE #8315: Allow running Joplin Server in debug mode

### PR DESCRIPTION
**Pull request by pedr**
Resolves https://github.com/laurent22/joplin/issues/8315

Using a debugger sometimes is much easier than trying to just print things to understand a bug, on this PR I'm suggesting a configuration that makes easy for everyone to run the serve project in debug mode inside vscode.

In the debug screen you can select "Server" and hit the play button to start the process. 

image

Before running the server there is a task called `transpile-server` that runs `yarn tsc` to make sure that the latest changes in the `package/server` are being executed. 

https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L360-L365

I think the "real" solution would be for this `transpile-server` task to be in the `tasks.json` file (inside the `tasks.json` we can use a `type` `npm` for the task that is more ergonomic), but I could not make it work like that, so I went with the simpler solution.

If there is any new environment variables that need to be included before execution, there is a place where they can be added:

https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L387-L391

Things that we could do to improve the experience is adding a build process to other projects that are used by the server project, but this still can be done with other terminals open.

This is my second time creating a configuration like this, so any feedback is welcomed!

### References:
[vscode debugger](https://code.visualstudio.com/docs/editor/debugging)
[Launch option to vscode workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces#_debugging)
[vscode tasks](https://code.visualstudio.com/docs/editor/tasks)

### CODE REVIEW

**Comment by pedr** [comment]
More images of it, for people who never used:

image
image
image

**Comment by laurent22** [comment]
Thanks Pedro, could you also add a document for it in readme/spec/server_debug.md? Basically copy what you have here to and add it to that file, so that we can find it back more easily later on

**Comment by pedr** [comment]
@laurent22 added the documentation file

File: readme/spec/server_debug.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+If there is any new environment variables that need to be included before execution, there is a place where they can be added:
+
+https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L387-L391
+
+Things that we could do to improve the experience is adding a build process to other projects that are used by the server project, 
+but this still can be done with other terminals open.
+
+### Modifying the configuration:
+
+To modify the configuration, for now, it is all in the `jopli.code-workspace` file, but if the needed we could also break
```
Comment (Daeraxa) [inline readme/spec/server_debug.md]:
```suggestion
To modify the configuration, for now, it is all in the `joplin.code-workspace` file, but if needed we could also break
```

File: readme/spec/server_debug.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Before running the server there is a task called `transpile-server` that runs `yarn tsc` to make sure that the 
+latest changes in the `package/server` are being executed. 
+
+https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L360-L365
+
+In a optiomal solution we would be doing this `transpile-server` task in the `tasks.json` file 
+(inside the `tasks.json` we can use a `type` `npm` for the task that is more ergonomic)
+
+If there is any new environment variables that need to be included before execution, there is a place where they can be added:
```
Comment (Daeraxa) [inline readme/spec/server_debug.md]:
```suggestion
If there are any new environment variables that need to be included before execution, there is a place where they can be added:
```

File: readme/spec/server_debug.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Debugging Server project with vscode
+
+Using a debugger sometimes is much easier than trying to just print things to understand a bug, 
+for the server project we have a configuration that makes easy for everyone to run in debug mode inside vscode.
```
Comment (Daeraxa) [inline readme/spec/server_debug.md]:
```suggestion
for the server project we have a configuration that makes it easy for everyone to run in debug mode inside vscode.
```

File: readme/spec/server_debug.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+In the debug screen you can select "Server" and hit the play button to start the process. 
+
+![image](https://github.com/laurent22/joplin/assets/5051088/cda6e3d6-32f3-4997-bd16-8413a6c962c5)
+
+Before running the server there is a task called `transpile-server` that runs `yarn tsc` to make sure that the 
+latest changes in the `package/server` are being executed. 
+
+https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L360-L365
+
+In a optiomal solution we would be doing this `transpile-server` task in the `tasks.json` file
```
Comment (Daeraxa) [inline readme/spec/server_debug.md]:
```suggestion
In a optimal solution we would be doing this `transpile-server` task in the `tasks.json` file 
```

File: readme/spec/server_debug.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L360-L365
+
+In a optiomal solution we would be doing this `transpile-server` task in the `tasks.json` file 
+(inside the `tasks.json` we can use a `type` `npm` for the task that is more ergonomic)
+
+If there is any new environment variables that need to be included before execution, there is a place where they can be added:
+
+https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L387-L391
+
+Things that we could do to improve the experience is adding a build process to other projects that are used by the server project,
```
Comment (Daeraxa) [inline readme/spec/server_debug.md]:
```suggestion
Something that we could do to improve the experience is adding a build process to other projects that are used by the server project, 
```

File: readme/spec/server_debug.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+https://github.com/laurent22/joplin/blob/4f37e6073ad549fe72206e2a125c27174b5d96a9/joplin.code-workspace#L387-L391
+
+Things that we could do to improve the experience is adding a build process to other projects that are used by the server project, 
+but this still can be done with other terminals open.
+
+### Modifying the configuration:
+
+To modify the configuration, for now, it is all in the `jopli.code-workspace` file, but if the needed we could also break
+the configuration in two files, one for the `launch.json` and other for the `tasks.json`.
```
Comment (Daeraxa) [inline readme/spec/server_debug.md]:
```suggestion
the configuration into two files, one for the `launch.json` and other for the `tasks.json`.
```

**Review by Daeraxa — COMMENTED** [review]
Just added some suggestions for grammar and spelling

**Comment by pedr** [comment]
@Daeraxa thanks for catching these mistakes, I should have checked any obvious mistakes in another text editor before submitting.

**Pull request merged**